### PR TITLE
Non story task: Add rapidAPI calls

### PR DESCRIPTION
**Pull request by dlhxu**
## Description
This PR is to add functionality to make external calls to the unofficial netflix api from rapidapi. Specifically, a single function to get the top 10 movies in Canada is provided, which makes a call to the "Advanced Search" endpoint with parameters on rapidapi set to: 

enfrate: 5
snfrate: 4
eimdbrate: 10
subtitle: Any
sortby: Relevance
syear: 1900
simdbrate: 6
audio: Any
eyear: 2020
vtype: Movie
genreid: 0
page: 1
clist: 33 (Corresponds to Canada country code)
query: (blank)
imdbvotes: gt100
andor: and

Note that these parameters do not directly correlate to actual query parameters used for the API call, and have to be entered into the rapidapi platform where a code snippet with properly formatted query parameters can be accessed.

For this sprint, only a single function and API call is necessary, with more flexible API calls being planned for future sprints.

## Changes
- Add rapidapi calls and wrap response as a promise
- fetch top 10 movies to display out of promise and emit to frontend after api call completes.

## Dependencies
Add axios, dotenv to make api calls to rapidapi

## Testing
- Unit tests available in rapidapiClient.test.js
- Manual integration testing to make sure that redirecting to swiping page still works, checking to make sure frontend receives array of objects

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -54,7 +57,12 @@ io.on('connection', (socket) => {
     // received signal to start a session from the host of a room, emit redirect signal to all guests and host
     socket.on('begin', ( user , callback) => {
         console.log('begin session signal received')
-        io.to(user.room).emit('sessionMembers', { room: user.room, users: getUsersInRoom(user.room), host: getHost(user.room)});
+
+        // get list of top 10 movies to send to all guests of the room
+        console.log('getting top 10')
```
Comment (varoongupta) [inline server/index.js:62]: Should the redis initializeRoom event be triggered prior to moving users to swiping page?

Comment (dlhxu) [reply server/index.js:62]: The initialize room event can be called in this .then block by passing in the returned top 10 list of movies , however it isn't in main yet. I'll add a comment indicating a good place to call the redis initializeRoom function in this code block!

File: server/external/rapidapiClient.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    
+    console.log('calling')
+    
+    return axios.request(options).then((response) => {
+      console.log('received response, formatting')
+      top10 = response.data.ITEMS.slice(-10)
+      console.log(top10)
+      return top10
+    }).catch(function (error) {
+      console.error(error);
```
[the remark this answers is not in the document]
Comment (dlhxu) [reply server/external/rapidapiClient.js:35]: Great catch, I've added the changes in the most recent commit

**Review by varoongupta — APPROVED** [review]

**Review by billxsheng — APPROVED** [review]
LGTM!

**Review by jeffreyquiambao — APPROVED** [review]
looks good to me!

File: server/index.js [hunk 3]
```diff
@@ -15,6 +17,7 @@ app.use(router);
 app.use(cors());
 
 const { addUser, addHost, removeUser, getUsersInRoom, getHost } = require('./users');
+const { getTop10 } = require('./movies');
```
Comment (rosakng) [inline server/index.js:20]: nicee 👍👍

**Review by rosakng — APPROVED** [review]
Great work with the api configurations Derek!!!

File: server/index.js [hunk 4]
```diff
@@ -54,7 +57,16 @@ io.on('connection', (socket) => {
     // received signal to start a session from the host of a room, emit redirect signal to all guests and host
     socket.on('begin', ( user , callback) => {
         console.log('begin session signal received')
-        io.to(user.room).emit('sessionMembers', { room: user.room, users: getUsersInRoom(user.room), host: getHost(user.room)});
+
+        // get list of top 10 movies to send to all guests of the room
+        console.log('getting top 10')
+        getTop10().then((top10) => {
+            // TODO add call to redis to initialize swiping room here
+            io.to(user.room).emit('sessionMembers', { room: user.room, users: getUsersInRoom(user.room), host: getHost(user.room), top10: top10});
```
Comment (rosakng) [inline server/index.js:65]: did you also alter the socket.on function to be expecting this top10 parameter?

Comment (dlhxu) [reply server/index.js:65]: yes I did, the socket.on function in lobby.js will expect this top10 parameter :)

Comment (rosakng) [reply server/index.js:65]: awesome LGTM!

**Pull request merged**